src/includes/class-alma-wc-payment-upon-trigger.php

```diff
@@ -0,0 +1,176 @@
+<?php
+/**
+ * Alma Payment Upon Trigger
+ *
+ * @package Alma_WooCommerce_Gateway
+ */
+
+use Alma\API\RequestError;
+
+if ( ! defined( 'ABSPATH' ) ) {
+	die( 'Not allowed' ); // Exit if accessed directly.
+}
+
+/**
+ * Alma_WC_Payment_Upon_Trigger
+ */
+class Alma_WC_Payment_Upon_Trigger {
+
+	/**
+	 * Logger
+	 *
+	 * @var Alma_WC_Logger
+	 */
+	private $logger;
+
+	/**
+	 * __construct.
+	 */
+	public function __construct() {
+		$this->logger = new Alma_WC_Logger();
+	}
+
+	/**
+	 * Callback function for the event "order status changed".
+	 *
+	 * @param integer $order_id The order id.
+	 * @param string  $previous_status Order status before it changes.
+	 * @param string  $next_status Order status affected to the order.
+	 * @return void
+	 * @SuppressWarnings("unused")
+	 */
+	public function woocommerce_order_status_changed( $order_id, $previous_status, $next_status ) { // phpcs:ignore Generic.CodeAnalysis.UnusedFunctionParameter
+
+		$order = wc_get_order( $order_id );
+		if ( 'alma' !== $order->get_payment_method() ) {
+			return;
+		}
+
+		if ( 'yes' !== alma_wc_plugin()->settings->payment_upon_trigger_enabled ) {
```

Comment:
according to [this discussion](https://almapay.slack.com/archives/C01U8Q55DQV/p1647343512876579?thread_ts=1647339218.597099&cid=C01U8Q55DQV)

we don't want check if settings is enabled here (only on payment creation).

We have to store information into the payment and check if payment was created with trigger option instead of checking `payment_upon_trigger_enabled` from settings

Reply:
done.